lib/sycamore/sycamore/data/element.py

```diff
@@ -204,6 +200,16 @@ def tokens(self) -> Optional[list[dict[str, Any]]]:
     def tokens(self, tokens: list[dict[str, Any]]) -> None:
         self.data["tokens"] = tokens
 
+    @property
+    def text_representation(self) -> Optional[str]:
+        if not self.data["text_representation"] and self.data["table"]:
+            self.data["text_representation"] = self.data["table"].to_csv()
+        return self.data["text_representation"]
```

Comment:
I'd prefer to see the code more like the following.  I think the logic is clearer and there are fewer hash lookups.  It handles missing keys better.  Also, the walrus operator.
```
tr = self.data.get("text_representation")
if not isinstance(tr, str):
    if (tbl := self.data.get("table")):
        tr = tbl.to_csv()
	self.data["text_representation"] = tr
return tr
```

Reply:
Sure, I've included a version of this.